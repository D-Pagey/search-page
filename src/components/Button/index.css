.button-div {
  width: 100%;
}

.button {
  background-color: #50ff7d;
  border: 0;
  border-radius: 6px;
  font-size: .95rem;
  font-weight: bold;
  height: 40px;
  margin: 5px 0;
  padding: 10px;
  width: 100%;
}