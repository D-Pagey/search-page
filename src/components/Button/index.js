import React from 'react';
import { string, func } from 'prop-types';
import styled from 'styled-components';

const ButtonWrapper = styled.div`
  width: 100%;
  `;

const Btn = styled.button`
  background-color: #50ff7d;
  border: 0;
  border-radius: 6px;
  font-size: .95rem;
  font-weight: bold;
  height: 40px;
  margin: 5px 0;
  padding: 10px;
  width: 100%;

  @media (min-width: 768px) {
    width: initial;
  }
`;

export default function Button({ children, onClick }) {
  return (
    <ButtonWrapper>
      <Btn type="Btn" onClick={onClick}>{children}</Btn>
    </ButtonWrapper>
  );
}

Button.propTypes = {
  children: string,
  onClick: func,
};

Button.defaultProps = {
  children: '',
  onClick: () => {},
};
