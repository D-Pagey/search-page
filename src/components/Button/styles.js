import styled from 'styled-components';

export const ButtonWrapper = styled.div`
  width: 100%;
  `;

export const Btn = styled.button`
  background-color: #50ff7d;
  border: 0;
  border-radius: 6px;
  font-size: .95rem;
  font-weight: bold;
  height: 40px;
  margin: 5px 0;
  padding: 10px;
  width: 100%;

  @media (min-width: 768px) {
    width: initial;
  }
`;
