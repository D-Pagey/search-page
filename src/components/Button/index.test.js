import React from 'react';
import { shallow } from 'enzyme';
import Button from '.';

it('should render', () => {
  const wrapper = shallow(<Button>Test</Button>);
  expect(wrapper).toMatchSnapshot();
});
