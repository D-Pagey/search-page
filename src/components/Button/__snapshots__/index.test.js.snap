// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="button-div"
>
  <button
    className="button"
    onClick={null}
    type="button"
  >
    Test
  </button>
</div>
`;
